civis/civis-aqueduct-utils/civis_aqueduct_utils/github.py

```diff
@@ -42,14 +44,17 @@ def upload_file_to_github(
 
     BASE = "https://api.github.com"
 
-    # Get the sha of the previous version
+    # Get the sha of the previous version.
+    # Operate on the dirname rather than the path itself so we
+    # don't run into file size limitations.
     r = requests.get(
-        f"{BASE}/repos/{repo}/contents/{path}",
+        f"{BASE}/repos/{repo}/contents/{os.path.dirname(path)}",

```

Comment:
Is S3 still ok for this?

Reply:
Yes, this should be okay for uploading files from S3. This is working around a limitation in the GitHub API, not the S3 API, so I don't think there should be any problem on that end.

Reply:
Update, just tried it with a 4MB file on S3 and it worked.